### ISSUE TITLE
AsyncMirror and Dropbox fails when there are subfolders with file

### ISSUE THREAD

**Issue by FredrikAleksander** [jvilk/BrowserFS#276]
When using AsyncMirror and Dropbox, and there are subfolders with files, it fails during configure. In the error, it seems to try and fetch but uses the subfolder path twice ![errmsg](https://user-images.githubusercontent.com/14351056/62247443-22d8e300-b3e6-11e9-8806-ddf3c93de771.png)
 .

I'm running the latest code from the master branch

A simple way to reproduce this error is by using this code in a web page, waiting for it to write the changes, and reload the page.

```
var dbx = new Dropbox.Dropbox({ fetch, accessToken: 'REDACTED' });
BrowserFS.configure({
    fs: "AsyncMirror",
    options: {
        sync: { fs: "InMemory" },
        async: {
            fs: "Dropbox",
            options: {
                client: dbx
            }
        }
    }
}, function (err) {
    if (err) { console.log(err); return; }

    var fs = BrowserFS.BFSRequire('fs');
    if(!fs.existsSync('/subfolder')) {
    	fs.mkdirSync('/subfolder');
    	fs.writeFileSync('/subfolder/somefile.txt', 'hello world');
    }
});
```

**Comment by Faless** [comment]
I can confirm this issue, after some debugging, it seems the reason for this is that Dropbox returns the absolute path of the file when listing directories.
The `AsyncMirror` instead, seems to expect each entry to be the relative path to the listed directory.

I'm pretty new to the codebase, so I'm not sure which behaviour is supposed to be correct (I tend to lean towards the `AsyncMirror` expectations). I'll dig further.

**Comment by Faless** [comment]
> (I tend to lean towards the `AsyncMirror` expectations).

Yep, [official doc](https://jvilk.com/browserfs/1.4.1/interfaces/_core_file_system_.filesystem.html#readdir) says `where files is an array of the names of the files in the directory excluding '.' and '..'.`, so the correct behaviour is relative path. I'll look into a fix, hoping it does not break other Dropbox internals.

**Comment by dr-vortex** [comment]
@FredrikAleksander Does this bug still exist?

**Comment by dr-vortex** [comment]
Closing (stale). If you would like to reopen this issue, please do so by creating a new issue in the relevant repositories of @browser-fs
